src/software/amazon/ionschema/internal/util/Range.kt

```diff
@@ -0,0 +1,75 @@
+package software.amazon.ionschema.internal.util
+
+import software.amazon.ion.IonInt
+import software.amazon.ion.IonList
+import software.amazon.ion.IonSymbol
+import software.amazon.ion.IonValue
+import software.amazon.ionschema.InvalidSchemaException
+
+/**
+ * Enum representing the support types of ranges.
+ */
+internal enum class RangeType {
+    INT,
+    INT_NON_NEGATIVE,
+    ION_NUMBER,
+    ION_TIMESTAMP,
+    ION_TIMESTAMP_PRECISION,
+}
+
+/**
+ * Interface for all range implementations.
+ */
+internal interface Range<in T : Any> {
+    fun contains(value: T): Boolean
+}
+
+/**
+ * Factory method for instantiating Range<T> objects.
+ */
+internal class RangeFactory {
+    companion object {
+        fun <T : Any> rangeOf(ion: IonValue, rangeType: RangeType): Range<T> {
```

Comment:
I think `: Any` in `<T: Any>` isn't doing anything.

Reply:
agreed, removed